tests/test_distributedarray_nccl.py

```diff
@@ -0,0 +1,438 @@
+"""Test the DistributedArray class
+Designed to run with n processes
+$ mpiexec -n 10 pytest test_distributedarray.py --with-mpi
+
+This file employs the same test sets as test_distributedarray under NCCL environment
+"""
+
+import numpy as np
+from mpi4py import MPI
+import pytest
+from numpy.testing import assert_allclose
+
+from pylops_mpi import DistributedArray, Partition
+from pylops_mpi.DistributedArray import local_split
+
+import cupy as cp
+import cupy.cuda.nccl as nccl
+import os
+
+np.random.seed(42)
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+host = MPI.Get_processor_name()
+
+
+# Use MPI to set-up NCCL Communication
+def initialize_nccl_comm():
+    local_rank = int(
+        os.environ.get("SLURM_LOCALID")
+        or os.environ.get("OMPI_COMM_WORLD_LOCAL_RANK")
+        or rank % cp.cuda.runtime.getDeviceCount()
+    )
+    device_id = local_rank
+    cp.cuda.Device(device_id).use()
+
+    if rank == 0:
+        nccl_id = nccl.get_unique_id()
+    else:
+        nccl_id = None
+    nccl_id = comm.bcast(nccl_id, root=0)
+
+    nccl_comm = nccl.NcclCommunicator(size, nccl_id, rank)
+    return nccl_comm
+
+
+nccl_comm = initialize_nccl_comm()
+
+# NCCL Does not support complex type
```

Comment:
Interesting, I wasn't aware of this... @hongyx11 any idea if this is in the roadmap of NCCL?